### PR TITLE
[Functions] (refactor): Various changes from review #3

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol [hunk 1]
```diff
@@ -225,52 +224,56 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
...
-    int256 weiPerUnitLink;
-    weiPerUnitLink = getFeedData();
-    if (weiPerUnitLink <= 0) {
-      revert InvalidLinkWeiPrice(weiPerUnitLink);
-    }
     uint256 executionGas = s_config.gasOverheadBeforeCallback + s_config.gasOverheadAfterCallback + callbackGasLimit;
+
     uint256 gasPriceWithOverestimation = gasPrice +
       ((gasPrice * s_config.fulfillmentGasPriceOverEstimationBP) / 10_000);
+    // @NOTE: Basis Points are 1/100th of 1%, divide by 10_000 to bring back to original units
```
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol:257]: Good comment.

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol [hunk 2]
```diff
@@ -11,29 +11,26 @@ import {FunctionsResponse} from "./libraries/FunctionsResponse.sol";
...
+  // s_totalLinkBalance tracks the total LINK sent to/from
   // this contract through onTokenTransfer, cancelSubscription and oracleWithdraw.
   // A discrepancy with this contract's LINK balance indicates that someone
   // sent tokens using transfer and so we may need to use recoverFunds.
-  uint96 private s_totalBalance;
+  uint96 private s_totalLinkBalance;
 
   // link token address
-  LinkTokenInterface private s_linkToken;
+  SafeERC20 private s_linkToken;
```
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol:33]: Missing import?

**Review by KuphJr — DISMISSED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol [hunk 3]
```diff
@@ -5,25 +5,33 @@ import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
...
   uint16 public constant MAX_CALLBACK_RETURN_BYTES = 4 + 4 * 32;
 
+  mapping(bytes32 id => address routableContract) private s_route;
+
+  error RouteNotFound(bytes32 id);
+
+  // Use empty bytes to self-identify, since it does not have an id
+  bytes32 private constant ROUTER_ID = bytes32(0);
+
   event RequestStart(
```
Comment (pinebit) [inline contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:35]: There is an ask from FE to include "estimated total cost in juels" to RequestStart event.

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:35]: Added!

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol:35]: Is there a world where we just want the whole commitment struct emitted here?

**Review by pinebit — DISMISSED** [review]

**Review by pinebit — APPROVED** [review]

**Review by pinebit — APPROVED** [review]

**Pull request merged**